File: frontend/src/lib/types/subscription.ts
export interface Subscription {
	id: string;
	user_id: string;
	status: 'active' | 'pending' | 'on-hold' | 'cancelled' | 'expired' | 'pending-cancel' | 'trial';
	plan: SubscriptionPlan;
	trial_ends_at: string | null;
	current_period_start: string | null;
	current_period_end: string | null;
	cancelled_at: string | null;
	paused_at: string | null;
	expires_at: string | null;
	payment_method: string | null;
	amount_paid: number;
	billing_cycles_completed: number;
	notes: string | null;
	created_at: string;
	updated_at: string;

	// Helper fields for UI
	productName?: string; // Mapped from plan.name
	price?: number; // Mapped from plan.price
	interval?: string; // Mapped from plan.billing_period
	nextPaymentDate?: string; // Mapped from current_period_end
}

export interface SubscriptionPlan {
	id: string;
	name: string;
	slug: string;
	description: string | null;
	price: number;
	billing_period: string;
	billing_interval: number;
	trial_days: number;
	currency: string;
	is_active: boolean;
}

export interface SubscriptionStats {
	total_subscriptions: number;
	active_subscriptions: number;
	monthly_revenue: number;
	churn_rate: number;
}

File: backend-rust/src/models/subscription.rs
//! Subscription Model
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use uuid::Uuid;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct UserSubscription {
    pub id: Uuid,
    pub user_id: Uuid,
    pub plan_id: Uuid,
    pub status: String,
    pub price: f64,
    pub currency: String,
    pub billing_period: String,
    pub start_date: DateTime<Utc>,
    pub next_payment_date: Option<DateTime<Utc>>,
    pub trial_end_date: Option<DateTime<Utc>>,
    pub cancelled_at: Option<DateTime<Utc>>,
    pub paused_at: Option<DateTime<Utc>>,
    pub stripe_subscription_id: Option<String>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Clone)]
pub struct SubscriptionWithPlan {
    pub id: Uuid,
    pub plan_id: Uuid,
    pub plan_name: String,
    pub plan_description: Option<String>,
    pub status: String,
    pub price: f64,
    pub currency: String,
    pub billing_period: String,
    pub start_date: DateTime<Utc>,
    pub next_payment_date: Option<DateTime<Utc>>,
    pub created_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct SubscriptionPlan {
    pub id: Uuid,
    pub name: String,
    pub description: Option<String>,
    pub price: f64,
    pub currency: String,
    pub billing_period: String,
    pub trial_days: i32,
    pub is_active: bool,
}
